### PR TITLE
feat: Convert serverless config to TypeScript

### PR DESCRIPTION
**Pull request by peterstarling**
As per: https://trello.com/c/S5SCkWK5/1134-convert-serverlessjs-file-to-typescript

Also cleans up the config file a bit by removing unnecessary template literals etc.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -7,10 +7,10 @@
     "emitDeclarationOnly": true,
     "target": "es2015",
 
-    "lib": ["esnext"],
+    "lib": ["esnext", "dom"],
```
Comment (jeysal) [inline tsconfig.json:10]: This is not correct for the majority of packages, and almost all packages inherit from this config.

File: tsconfig.json [hunk 2]
```diff
@@ -26,6 +26,11 @@
 
     "noErrorTruncation": true
   },
-  "files": [],
-  "include": []
+  "include": [
```
Comment (jeysal) [inline tsconfig.json:29]: `apps` should not be in the root build, there are `tsconfig`s specifically for them

**Review by jeysal — CHANGES_REQUESTED** [review]
The `tsconfig` changes made to support using `ts-node` here are not correct. Please just use `babel-node` instead, it makes things a lot easier :see_no_evil:

File: tsconfig.json (hunk 1 above)
Comment (peterstarling) [reply tsconfig.json]: yah that was dumb, sorry for being a n00b :cry:

File: tsconfig.json (hunk 2 above)
Comment (peterstarling) [reply tsconfig.json]: yeah thats fine, i dont think i actually wanted to check that in, its not even required for the serverless build

**Pull request closed without merging**